NuRadioReco/detector/RNO_G/rnog_detector.py

```diff
@@ -0,0 +1,1488 @@
+
+    def get_time_delay(self, station_id, channel_id, cable_only=False, use_stored=False):
+        """ Return the sum of the time delay of all components in the signal chain calculated from the phase
+
+        Parameters
+        ----------
+
+        station_id: int
+            The station id
+
+        channel_id: int
+            The channel id
+
+        cable_only: bool
+            If True: Consider only cables to calculate delay. (Default: False)
+
+        use_stored: bool
+            If True, take time delay as stored in DB rather than calculated from response. (Default: False)
+
+        Returns
+        -------
+
+        time_delay: float
+            Sum of the time delays of all components in the signal chain for one channel
+        """
+
+        signal_chain_dict = self.get_channel_signal_chain(
+            station_id, channel_id)
+
+        time_delay = 0
+        for key, value in signal_chain_dict["response_chain"].items():
+
+            if re.search("cable", key) is None and cable_only:
+                continue
+
+            if use_stored:
+                if "time_delay" not in value or "cable_delay" not in value:
+                    self.logger.warning(
+                        f"The signal chain component \"{key}\" of station.channel "
+                        f"{station_id}.{channel_id} has no cable/time delay stored... Skip it")
+                    continue
+
+                try:
+                    time_delay += value["time_delay"]
+                except KeyError:
+                    time_delay += value["cable_delay"]
+
+            else:
+                ydata = [value["mag"], value["phase"]]
+                response = Response(value["frequencies"], ydata, value["y-axis_units"],
+                                    name=key, station_id=station_id, channel_id=channel_id)
+
+                time_delay += response._get_time_delay()
+
+        return time_delay
+
+
+    def get_site(self, station_id):
+        """
+        This detector class is exclusive for the RNO-G detector at Summit Greenland.
+
+        Returns
+        -------
+
+        site: str
+            Returns "summit"
+        """
+        return "summit"
+
+    def db(self):
+        """ Temp. interface for development """
+        return self.__db
+
+    def set_detector_time(self, time):
+        """ Temp. interface for development """
+        self.__set_detector_time(time)
+
+
+class Response:
+    """
+    This class provides an interface to read-in and apply the complex response functions of the
+    various components of the signal chain of a RNO-G channel. The class allows to combine the
+    response of several components into one response by multiplying them.
+    """
+
+    def __init__(self, frequency, y, y_unit, time_delay=0, weight=1,
+                 name="default", station_id=None, channel_id=None,
+                 remove_time_delay=True, debug_plot=True):
```

Comment:
debug_plot should probably (as claimed by the docstring) be set to `False` by default